src/core/qgsdistancearea.cpp

```diff
@@ -537,6 +540,111 @@ double QgsDistanceArea::measureLine( const QgsPoint &p1, const QgsPoint &p2, Qgs
   return result;
 }
 
+double QgsDistanceArea::measureLineProjected( const QgsPoint &p1, double distance, double azimuth, QgsPoint *projected_point )
+{
+  double result = 0.0;
+  QgsPoint p2;
+  if ( geographic() )
```

Comment:
Yes, it is very clumsy and not clear
* I assumed at first that the default being Qsg84, that it would be usable as such

How about adding constructors with
* QgsCoordinateReferenceSystem
* String : 'EPSG:xxxx'
* long srid

and when Geographic, call setEllipsoid.

This would be more straight forward.